### PR TITLE
Remove fake testing table info builder 

### PR DESCRIPTION
**Pull request by seut**
Instead, real table infos using the SQLExecutor are build making
unit test relying on existing table infos “real world” scenarios.

This also eliminates the need to maintain a fake table info builder
which already lead to some wrongly built tables, see test adjustments.

First commit removes an artificial limitation to allow casts to ``SET`` which was required as 
real table infos cannot contain a column of type ``SET`` but some (internal) scalars required it as an argument and could otherwise not be unit tested.

### CODE REVIEW

File: sql/src/test/java/io/crate/testing/SQLExecutor.java [hunk 1]
```diff
@@ -570,6 +582,28 @@ public static Builder builder(ClusterService clusterService, int numNodes, Rando
         return new Builder(clusterService, numNodes, random);
     }
 
+    public static DocTableInfo tableInfo(RelationName name, String stmt, ClusterService clusterService) {
+        try {
+            SQLExecutor executor = builder(clusterService)
+                .addTable(stmt)
+                .build();
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/testing/SQLExecutor.java]: The builder creation and build altogether is fairly expensive, I wonder if we should mark these methods somehow to indicate that if more than 1 table is required it is recommended to create the executor only once?

E.g in T3 there are 2 loops that use this method inside the loops.

File: sql/src/test/java/io/crate/testing/T3.java [hunk 2]
```diff
@@ -22,103 +22,104 @@
...
+    public static final RelationName T5_RN = new RelationName(Schemas.DOC_SCHEMA_NAME, "t5");
+
+    private static final Map<RelationName, String> RELATION_DEFINITIONS = ImmutableMap.of(
+        T1_RN, T1_DEFINITION,
+        T2_RN, T2_DEFINITION,
+        T3_RN, T3_DEFINITION,
+        T4_RN, T4_DEFINITION,
+        T5_RN, T5_DEFINITION);
+
+    public static ImmutableList<AnalyzedRelation> relations(ClusterService clusterService) {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/testing/T3.java]:
```suggestion
    public static List<AnalyzedRelation> relations(ClusterService clusterService) {
```

To not expose implementation detail across packages/classes

File: sql/src/test/java/io/crate/analyze/InsertFromValuesAnalyzerTest.java [hunk 3]
```diff
@@ -73,105 +67,72 @@
...
-        TestingTableInfo.Builder generatedNestedClusteredByInfo = new TestingTableInfo.Builder(
-            generatedNestedClusteredByRelationName, SHARD_ROUTING)
-            .add("o",
-                ObjectType.builder().setInnerType("serial_number", DataTypes.INTEGER).build(),
-                null,
-                ColumnPolicy.DYNAMIC)
-            .addGeneratedColumn("routing_col", DataTypes.INTEGER, "o['serial_number'] + 1", false)
-            .clusteredBy("routing_col");
-        executorBuilder.addDocTable(generatedNestedClusteredByInfo);
+            .addTable(
```
Comment (mkleen) [inline sql/src/test/java/io/crate/analyze/InsertFromValuesAnalyzerTest.java]: The new version of the table definition is much more readable, although the old builder style version is probably a lot easier to maintain.

File: sql/src/test/java/io/crate/testing/SQLExecutor.java (hunk 1 above)
Comment (seut) [reply sql/src/test/java/io/crate/testing/SQLExecutor.java]: I've added method documentation and also removed the T3 usages.

**Pull request merged**